flexget/plugins/sites/torrentz2.py

```diff
@@ -0,0 +1,121 @@
+        urls = torrent_cache.infohash_urls(thash)
+        # default to first shuffled url
+        entry['url'] = urls[0]
+        entry['urls'] = urls
+        entry['torrent_info_hash'] = thash
+
+    def search(self, task, entry, config=None):
+        config = self.process_config(config)
+        feed = REPUTATIONS[config['reputation']]
+        entries = set()
+        for search_string in entry.get('search_strings', [entry['title']]):
+            query = normalize_unicode(search_string + config.get('extra_terms', ''))
+            for domain in ['eu', 'me', 'ch', 'in']:
+                # urllib.quote will crash if the unicode string has non ascii characters, so encode in utf-8 beforehand
+                url = 'http://torrentz2.%s/%s?q=%s' % (domain, feed, quote(query.encode('utf-8')))
+                log.debug('requesting: %s' % url)
+                try:
+                    r = task.requests.get(url)
+                    break
+                except requests.ConnectionError as err:
+                    # The different domains all resolve to the same ip, so only try more if it was a dns error
+                    log.warning('torrentz2.%s connection failed. Error: %s' % (domain, err))
+                    continue
+                except requests.RequestException as err:
+                    raise plugin.PluginError('Error getting torrentz2 search results: %s' % err)
+
+            else:
+                raise plugin.PluginError('Error getting torrentz2 search results')
+
+            if not r.content.strip():
+                raise plugin.PluginError('No data from %s. Maybe torrentz2 is blocking the FlexGet User-Agent' % url)
+
+            rss = feedparser.parse(r.content)
+
+            if rss.get('bozo_exception'):
+                raise plugin.PluginError('Got bozo_exception (bad rss feed)')
+
+            for item in rss.entries:
+                m = re.search(r'Size: ([\d]+) Mb Seeds: ([,\d]+) Peers: ([,\d]+) Hash: ([a-f0-9]+)',
+                              item.description, re.IGNORECASE)
+                if not m:
+                    log.debug('regexp did not find seeds / peer data')
+                    continue
+
+                entry = Entry()
+                entry['title'] = item.title
+                entry['url'] = item.link
+                entry['content_size'] = int(m.group(1))
+                entry['torrent_seeds'] = int(m.group(2).replace(',', ''))
+                entry['torrent_leeches'] = int(m.group(3).replace(',', ''))
+                entry['torrent_info_hash'] = m.group(4).upper()
+                entry['search_sort'] = torrent_availability(entry['torrent_seeds'], entry['torrent_leeches'])
+                entries.add(entry)
+
+        log.debug('Search got %d results' % len(entries))
+        return entries
+
+
+@event('plugin.register')
+def register_plugin():
+    plugin.register(Torrentz2, 'torrentz2', groups=['urlrewriter', 'search'], api_ver=2)
```

Comment:
The 'groups' argument was deprecated, this should be replaced with 'interfaces'. (This will also clear up the problem with tests where it's being considered a task plugin.)